go/vt/topo/locks.go

```diff
@@ -120,6 +126,28 @@ type locksKeyType int
 
 var locksKey locksKeyType
 
+// Support different lock types.
+type LockType int
+
+const (
+	Blocking LockType = iota
+	NonBlocking
+	Named
+)
+
+func (lt LockType) String() string {
+	switch lt {
+	case Blocking:
+		return "blocking"
+	case NonBlocking:
+		return "non blocking"
+	case Named:
+		return "named"
+	default:
+		return "unknown"
+	}
```

Comment:
Yeah, I went back and forth on that. An iota is an int so you can specify "invalid" values which then are ignored and a default is applied. Should we show that in the String representation as well? I'm fine either way (for many things we don't even implement Stringer so it would show you "6" eg.

Reply:
I think it will be less confusing to keep the defaults the same.

Reply:
I've addressed that here: https://github.com/vitessio/vitess/pull/16260/commits/8de719a542967378ad632634e172cd5d7bf7fbfd